### DISCUSSION TITLE
[Update|Insert|Delete]WithOutputIntoOutput

### DISCUSSION THREAD

**Discussion by croban** [linq2db/linq2db#3831] [Ideas]
Hi 
[Update|Insert|Delete]WithOutput and Merge API are really GREAT feature and nice things can be made with it.
Only thing what I miss, and I would like to be able to do is something like this (WithOutputInto+Output):

```
update [Order] 
	set Somehing='XA'...
output inserted.* into OrderAudit
output inserted.*
	where ...
```

It would be useful for creating Audit entries (basically copy of main table) and getting results out of it in same round trip.
Otherwise, to get new created/updated entities where calculated columns are present inside main table I have to do following [Insert|Update]WithOutput and then in second query I can do BulkCopy of those results and to be safe I have to encapsulate these two steps inside Transaction. 
Triggers cannot be used because usage of OUTPUT without INTO is forbidden (see [this](https://github.com/linq2db/linq2db/issues/2982)), also this [Insert|Update|Delete]WithResultFrom Triggers could be nice feature.
Same Feature for MergeWithOutputInto+Output would be great!

**Comment by MaceWindu** [comment]
Didn't know it is possible. I've created [feature request](https://github.com/linq2db/linq2db/issues/3832), but frankly speaking we currently don't have much time to work on it, so PR is appreciated
